specification/security/suggest_user_profiles/types.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UserProfileId } from '@security/_types/UserProfile'
+import { SingleKeyDictionary } from '@spec_utils/Dictionary'
+
+export class Hint {
+  /**
+   * A list of Profile UIDs to match against.
+   */
+  uids?: UserProfileId[]
+  /**
+   * A single key-value pair to match against the labels section
+   * of a profile. A profile is considered matching if it matches
+   * at least one of the strings.
+   */
+  labels?: SingleKeyDictionary<string, HintLabelValue>
+}
+
+export type HintLabelValue = string | string[]
```

Comment:
Not sure we need dedicated types to express array leniency, even more for a one-time use.

Ultimately, we should introduce a type for lenient arrays similar to the recent `Stringified`, i.e. `type LenientArray<T> = T | T[]`. Or just bite the bullet and have array leniency be an implicit contract throughout the entire spec since every array will have to be a `LenientArray`.

Reply:
I'll inline this `string | string[]`. Wasn't sure if having this inline would be a problem for generators.